### PR TITLE
new couchdb tools in tools, new check tools in pca_tools

### PR DESCRIPTION
**Pull request by MRiganSUSX**

### CODE REVIEW

**Comment by mnirkko** [comment]
A few questions here:
- Aren't the get_channel and update_channel scripts already contained in PR #47 ?
- Do we want PCA run lists to be contained in the Git repo? I have no problem with that, but please use YYYY-MM format (not single digit months).
Other than that, looks good!

File: pca_tools/check_couch.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+	# load run, connect to db
+	for run in content:
+		print run
+
+		for item in db.view('_design/runs/_view/run_by_number', startkey=int(run), endkey=int(run)):
+    			id = item.id
+			subruns = len(db[id]['sub_run_info'])
+			print "Subruns in couchdb: ", subruns
+			if (subruns != 40):
```
Comment (mnirkko) [inline pca_tools/check_couch.py]: Not sure we want this hard-coded?

File: pca_tools/check_couch.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import os
+import couchdb
+import getpass
+
+# connect to couch
+pw = getpass.getpass("CouchDB password: ")
+couchserver = couchdb.Server("http://snoplus:%s@couch.snopl.us" % pw)
```
Comment (mnirkko) [inline pca_tools/check_couch.py]: There should perhaps be a config file from which these hard-coded settings are loaded. Compare to tellie.cfg in the base directory...

File: pca_tools/check_lists.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+for list in lists:
+	missing = []
+	with open(list) as f:
+    		content = f.readlines()
+		content = [x.strip() for x in content] 
+		max_run = max(content)
+		min_run = min(content)
+		
+		# check whether the file is stored at feyman
+		path = "/lustre/scratch/epp/neutrino/snoplus/TELLIE/TELLIE_PCA_RUNS"
```
Comment (mnirkko) [inline pca_tools/check_lists.py]: Belongs in a config file...

File: pca_tools/check_lists.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+		
+		for run in content:
+			file_string = "SNOP_0000" + run + "_000.zdab"
```
Comment (mnirkko) [inline pca_tools/check_lists.py]: Not sure this is the default filename, but either way it should not be hardcoded this way.

File: tools/get_channel.py [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+# text formatting
+matplotlib.rcParams.update({'font.size': 13})
+
+# set up access to database
+database = tellie_database.TellieDatabase('http://couch.snopl.us', 'telliedb')
+
+# load channel data from couchDBs
+def get_channel(channel):
+    # key for query
+    key = [ int(channel), [ 10870, 2147483647 ], 0, -2 ]
```
Comment (mnirkko) [inline tools/get_channel.py]: Is this a range? Should perhaps belong in a config file too?

File: tools/update_channel.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    # ask for new data
+    channel = raw_input('Channel (INT): ')
+    cone = raw_input('Cone: ')
+    driver = raw_input('Driver (INT): ')
+    PIN_board = raw_input('PIN board: ')
+    fibre_delay = raw_input('Fibre delay (FLOAT): ')
+    print channel, cone, driver, PIN_board, fibre_delay
+
+    # get the correct doc
+    key = [ int(channel), [ 10870, 2147483647 ], 0, -2 ]
```
Comment (mnirkko) [inline tools/update_channel.py]: Hard-coded range.

**Review by mnirkko — CHANGES_REQUESTED** [review]
On second viewing, there are a few things that could be improved. Please have a look @MRiganSUSX

File: pca_tools/check_couch.py (hunk 1 above)
Comment (MRiganSUSX) [reply pca_tools/check_couch.py:38]: taken to config

File: pca_tools/check_couch.py (hunk 2 above)
Comment (MRiganSUSX) [reply pca_tools/check_couch.py:7]: taken to config

File: pca_tools/check_lists.py (hunk 3 above)
Comment (MRiganSUSX) [reply pca_tools/check_lists.py:20]: taken to config

File: pca_tools/check_lists.py (hunk 4 above)
Comment (MRiganSUSX) [reply pca_tools/check_lists.py:23]: modified to string formating

File: tools/get_channel.py (hunk 5 above)
Comment (MRiganSUSX) [reply tools/get_channel.py:30]: this spans the start of the tellie use and end of tellie tables, it's not an issue of this small piece of code but deeper sno+ setting, will leave like this for now (should not ever change)

File: tools/update_channel.py (hunk 6 above)
Comment (MRiganSUSX) [reply tools/update_channel.py:33]: same as above

**Comment by MRiganSUSX** [comment]
Added most of the requested changes, left comments for the rest. Thanks for your time!

**Review by mnirkko — APPROVED** [review]
This looks good, no further comments on the contents.

**Comment by mnirkko** [comment]
There are two ODS files and I'm not sure of their purpose.
-  pca_tools/tellie_tables.ods just contains one line that points to a specific path on one of @MRiganSUSX computers.
- pca_tools/TELLIE_water_setpoints.ods is a full copy of the Google spreadsheet (I assume), is this intended as backup?

**Comment by MRiganSUSX** [comment]
Thanks for the comments and the review.
- will remove tellie_tables.ods from working tree
- the water_setpoints.ods is indeed a backup. It's important enough file that should be available at more than 1 location I think.

**Pull request merged**